ouroboros-consensus/src/Ouroboros/Consensus/HeaderStateHistory.hs

```diff
@@ -107,7 +107,7 @@ cast (HeaderStateHistory history) =
 -- rewinding that header state again by @j@ where @i + j > k@ is not possible
 -- and will yield 'Nothing'.
 rewind ::
-     forall blk. (BlockSupportsProtocol blk, HasAnnTip blk)
```

Comment:
Was there a bug in GHC <9.2 that caused it to be missing unnecessary constraints or something?

Reply:
Yes, in earlier GHCs that was incorrectly considered an necessary constraint. It never was necessary and with 9.2 it is flagged as redundant.